### PR TITLE
Lesser Summon Guns

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Summons a single shot bolt action rifle into your hand. Firing the rifle automatically tosses it aside, switches your hand, and summons a new single shot rifle. This continues for 30 rifles (it has to compete with laser eyes for a spell slot, after all).

:cl: Kor
rscadd: The wizard has a new spell, Lesser Summon Guns. This summons an unending stream of single shot bolt action rifles into his hands, automatically replacing themselves as you fire.
/:cl:

### CODE REVIEW

**Comment by Boredone** [comment]
Um, not 100% sure, but in checking how much damage the bolt actions would do, I think the ammo_casing/a762 they use got removed in #13629. Probobly cause thats what the SAW used, and they didn't think to check what else used it?
Either way, you'll probobly want to readd that.

**Comment by KorPhaeron** [comment]
The casing still exists, it does 35 damage now

**Comment by Boredone** [comment]
Ah, yeah, sorry. Wonder why they moved it from the ammo casing page to the saw page then though.

**Comment by AlexCrimson** [comment]
Jeez... when is Wizard getting some kind of "Gun Game" summon? I feel like we are pretty close.

**Comment by Supermichael777** [comment]
Madoka references? really?

**Comment by KorPhaeron** [comment]
@Supermichael777 

If it makes you feel any better, playing Vermintide and using the brace of pistols is what inspired this more than anything

**Comment by Cebasiz** [comment]
can i get a drill hairstyle and a dress on the wizard ship please

File: code/datums/spells/infinite_guns.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	action_icon_state = "bolt_action"
+
+
+
+/obj/effect/proc_holder/spell/targeted/infinite_guns/cast(list/targets, mob/user = usr)
+	for(var/mob/living/carbon/C in targets)
+		C.drop_item()
+		C.swap_hand()
+		C.drop_item()
+		var/obj/item/weapon/gun/projectile/shotgun/boltaction/enchanted/GUN = new /obj/item/weapon/gun/projectile/shotgun/boltaction/enchanted
```
Comment (duncathan) [inline code/datums/spells/infinite_guns.dm]: The path is implicit, you can just do ` = new `

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 2]
```diff
@@ -113,6 +113,34 @@
...
+	desc = "Careful not to lose your head."
+	var/guns_left = 30
+	mag_type = /obj/item/ammo_box/magazine/internal/boltaction/enchanted
+
+/obj/item/weapon/gun/projectile/shotgun/boltaction/enchanted/shoot_live_shot(mob/living/user as mob|obj, pointblank = 0, mob/pbtarget = null, message = 1)
+	..()
+	user.drop_item()
+	if(guns_left)
+		user.swap_hand()
+		var/obj/item/weapon/gun/projectile/shotgun/boltaction/enchanted/GUN = new /obj/item/weapon/gun/projectile/shotgun/boltaction/enchanted(src)
```
Comment (tkdrg) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: Same as above

**Comment by tkdrg** [comment]
Hilarious idea 10/10

**Comment by KorPhaeron** [comment]
@tkdrg 

Ready

**Pull request merged**